src/main/java/array/Array.java

```diff
@@ -0,0 +1,49 @@
+package array;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/**
+ * Interface for an array with undefined number of dimensions
+ *
+ * @param <T> - type of values that allow to put into array
+ * @param <C> - coordinates class for navigation through array
```

Comment:
It's very minor thing, but usually `T` & `R`/ `S` are used as second generic param of the type

Reply:
`C` is for `coordinates`. I prefer to keep it.

Reply:
ok